woocommerce/woocommerce
# [Product Block Editor]: update the attributes list optimistically when a new attribute is created

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR uses the `optimisticQueryUpdate` option to update the attributes list optimistically, when a new attribute is created form the new attribute modal.

Closes #https://github.com/woocommerce/woocommerce/issues/46716



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the Product Editor
2. Add/edit a new product
3. Go to the `Variations` section
4. Click on the `Add options` button
5. Confirm the attribute selector works as usual 
  * Create a new attribute
  * Select an existing attribute

Additionally, you can. check how the app propagates and populates the data when creating a new attribute

1. Open the network tab of your broser
2. Add/edit a new product
3. Go to the `Variations` section
4. Open the network tab of your browser dev tool
5. Clean the current requests
6. Click on the `Add options` button to open the modal
7. Confirm the performs a request to pull the attributes

<img width="697" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/00d1d540-47b5-4420-84fd-6aeff702a278">

8. Add a new attribute
9. Confirm the client does not perform a new request to pull the attributes
10. If you have installed the Redux dev tool, confirm how the app populates the store


https://github.com/woocommerce/woocommerce/assets/77539/9b976578-a847-408f-964a-6bc3e25bcdb7





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
